### ISSUE TITLE
cnaster defect: the two integer decoders report different ploidies for the same profile

### ISSUE THREAD

**Issue by michaelJwilson** [michaelJwilson/port#181]
**`hill_climbing_integer_copynumber_oneclone` and `_fixdiploid_milp` return the identical copy-number profile and disagree on the third return value — 4 against 3. `run_cnaster` logs it as "best ploidy" beside the loss. CalicoST's hill climbers return no ploidy at all, so the third value is `cnaster`'s addition.**

## Measured

On four planted integer states `(1,1), (2,1), (3,1), (2,2)` and their own observables, both decoders return those four pairs at an objective of 6.8e-16 and 6.7e-16 — machine zero — and report:

| decoder | copies | "best ploidy" |
| --- | --- | ---: |
| `_oneclone` | `((1,1), (2,1), (3,1), (2,2))` | **4** |
| `_fixdiploid_milp` | identical | **3** |

At most one of them is the ploidy of the answer, and the log line does not say which definition it searched over.

## CalicoST: two values, not three

```python
# find_integer_copynumber.py:162, :235
return best_integer_copies, best_obj
```

Both of CalicoST's climbers — `_fixdiploid` and `_oneclone` — return a pair. `max_medploidy` is a *search bound* there, not something reported back. `cnaster` added the third return and `_fixdiploid_milp` is a new variant with no CalicoST counterpart, so **the disagreement is `cnaster`'s and arrived with the MILP rewrite.**

## Consequence

```python
# run_cnaster.py:1420
logger.info(f"Solved for (max. med ploidy, clone) = ({max_medploidy}, {s}) with integer copy number loss = {loss:.4e} and best ploidy = {best_ploidy}")
```

The value is logged and not otherwise used, so nothing downstream is wrong today. What is wrong is the record: a run's log states a ploidy per clone, and which of two incompatible definitions it reports depends on whether `max_medploidy` was set — the branch that chooses between the two decoders. A reader comparing ploidy across runs is comparing two quantities.

## Pinned

`test_the_two_decoders_disagree_on_the_ploidy_of_the_same_answer` ([#173](https://github.com/michaelJwilson/port/pull/173), `warning`) asserts both the identical copies and the differing ploidies, so it fails if either the disagreement or the profiles change.

Resolving it needs the definition each searches over, which is a `cnaster` question; `integer_copy_numbers.tex` is what settles which one the paper means.

Refs #6, #25, #160, #173.
